java-frontend/src/main/java/org/sonar/java/se/ExplodedGraph.java

```diff
@@ -107,27 +116,50 @@ public String toString() {
     public Node parent;
     private final List<LearnedConstraint> learnedConstraints;
 
+    private final List<LearnedValue> learnedSymbols;
+
     Node(ProgramPoint programPoint, @Nullable ProgramState programState) {
       this.programPoint = programPoint;
       this.programState = programState;
       learnedConstraints = new ArrayList<>();
+      learnedSymbols = new ArrayList<>();
     }
 
     public void setParent(@Nullable Node parent) {
       this.parent = parent;
       if(parent != null) {
         programState.constraints.forEach((sv, c) -> {
           if(parent.programState.getConstraint(sv) != c) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)